Okta.Xamarin/Okta.Xamarin/IOktaStateManager.cs

```diff
@@ -0,0 +1,48 @@
+﻿// <copyright file="OktaState.cs" company="Okta, Inc">
+// Copyright (c) 2019-present Okta, Inc. All rights reserved.
+// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.
+// </copyright>
+
+using System;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace Okta.Xamarin
+{
+    /// <summary>
+    /// State manager interface.
+    /// </summary>
+    public interface IOktaStateManager
+    {
+        string AccessToken { get; }
+
+        IOktaConfig Config { get; set; }
+
+        DateTime? Expires { get; }
+
+        string IdToken { get; }
+
+        bool IsAuthenticated { get; }
+
+        HttpResponseMessage LastApiResponse { get; set; }
```

Comment:
I know this is for debug but it feels strange to have a HttpResponseMessage in the OktaStateManager. Are we sure we want to expose this to devs?

Reply:
I'll give this some thought to wrap it in a more `debug`ish kind of way.

Reply:
Moved logic to client as suggested